modules/tree/main/index.coffee

```diff
@@ -1,5 +1,35 @@
+createNodeView = (node, renderer, lazy) ->
+  node.view('.hx-tree-node')
+    .enter (d) -> @append(createTreeNode(d, renderer, lazy)).node()
+
+createChildren = (children, renderer, lazy) ->
+  children.map (child) -> createTreeNode(child, renderer, lazy)
+
+createTreeNode = (data, renderer, lazy) ->
```

Comment:
This function now handles the creation of the tree nodes when in JSON mode - it doesn't need to care about the tree because the logic for open/close is in `format` (to retain handling for `html` trees)